Release notes and version for v0.9.11

kristapsk opened the pull request:
Please review and test. Treat this as a release candidate. :)

MarnixCroes commented on docs/release-notes/release-notes-0.9.11.md:
```diff
@@ -0,0 +1,176 @@
+
+### Removal of Python 3.7 support, 3.8 or newer is required now
+
+Python 3.7 has been end of-life since June 2023 in it was blocker for some necessary dependency updates.
```
```suggestion
Python 3.7 has been end of-life since June 2023 and it was a blocker for some necessary dependency updates.
```

MarnixCroes commented on docs/release-notes/release-notes-0.9.11.md:
```diff
@@ -0,0 +1,176 @@
+
+Python 3.7 has been end of-life since June 2023 in it was blocker for some necessary dependency updates.
+
+Note that we currently only Python 3.8, 3.9, 3.10 and 3.11 are supported, not Python 3.12 and newer.
```
```suggestion
Note that currently only Python 3.8, 3.9, 3.10 and 3.11 are supported, not Python 3.12 and newer.
```

MarnixCroes commented on docs/release-notes/release-notes-0.9.11.md:
```diff
@@ -0,0 +1,176 @@
+
+### Performance improvements
+
+Lots of performance improvements made in wallet code that makes big impact especially with old wallets, with a lots of transaction history (in extreme cases time needed for some operations was improved from 40+ minutes to around 10 seconds).
```
```suggestion
Lots of performance improvements made in wallet code that makes big impact especially with old wallets, with a lots of transaction history (in extreme cases, the time needed for some operations was improved from 40+ minutes to around 10 seconds).
```

MarnixCroes commented on docs/release-notes/release-notes-0.9.11.md:
```diff
@@ -0,0 +1,176 @@
+
+Lots of performance improvements made in wallet code that makes big impact especially with old wallets, with a lots of transaction history (in extreme cases time needed for some operations was improved from 40+ minutes to around 10 seconds).
+
+This involves caching some computed data in wallet, for cases when if there is some bug with this discovered, new configuration option `wallet_caching_disabled` is added to `joinmarket.cfg`, which allows to disable it when syncing the wallet.
```
```suggestion
This involves caching some computed data in the wallet, in case a bug is discovered with this, a new configuration option `wallet_caching_disabled` will be added to `joinmarket.cfg`, which allows to disable it when syncing the wallet.
```

AdamISZ replied:
> It is added, not will be added.

Yep, probably should do the other part of @MarnixCroes 's edit here, though, i.e. "in case a bug is discovered".

MarnixCroes commented:
lgtm, few nits

AdamISZ commented:
LGTM at https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/1671/commits/dd462a613e94b30d5916e20258fbe5eada689f04

Though it's outside the norm, it would not be unreasonable to shout out @whitslack specifically for an extremely significant contribution here (in the section on wallet performance). Whatever you think, it's not a huge deal.
A vaguely related point: the description of how wallet syncing performance has improved is good, but I'm wondering if we could add this point (because it's such a common, big, problem for users): that if they are syncing a heavily used wallet from a seedphrase on a new Core instance, they can now use a large `-g` gaplimit and it will take far less time than it did before. I think your description of the ability to disable caching is fine. Another thing you could mention (maybe?) is that the size of the `*.jmdat` files will be larger than before, and that this is normal, because it contains cached data (but still encrypted).

kristapsk commented:
Addressed review comments by @MarnixCroes and @AdamISZ.

> Though it's outside the norm, it would not be unreasonable to shout out @whitslack specifically for an extremely significant contribution here (in the section on wallet performance). 

Added that to "Credits" section. Also mentioned new first time contributors.
